### ISSUE TITLE
Non-nullable field is generated as nullable kotlin property

### ISSUE THREAD

**Issue by masc3d** [jOOQ/jOOQ#12172]
### Expected behavior
for non-nullable / pk field:
```
    var id: Int
        set(value): Unit = set(0, value)
    @Nonnull
        get(): Int = get(0) as Int
```
### Actual behavior
nullable type is generated, not aligning with table field type and generated annotation:
```
    var id: Int?
        set(value): Unit = set(0, value)
    @Nonnull
        get(): Int? = get(0) as Int?
```

### Versions

- jOOQ: 3.14.12, 3.15.0
- Java: 16
- Database (include vendor): mysql-5.7

**Comment by lukaseder** [comment]
Thanks for your report.

See also: https://github.com/jOOQ/jOOQ/issues/10212

It will be important to keep in mind that whatever the `KotlinGenerator` produces must match the annotations, so I'm keeping this open.

**Comment by lukaseder** [comment]
https://github.com/jOOQ/jOOQ/issues/10212 has been implemented in jOOQ 3.18. Regarding the matching annotations / kotlin nullability, I'm not sure if this needs to be addressed. The kotlin nullability is opt-in for backwards compatibility, but probably, most kotlin users will turn it on by default?

I'm closing this as a duplicate of #10212 for now. If someone has a case why annotations should be generated in kotlin, but the flags shouldn't be turned on, I'll look into this again.
